src/ray/raylet/scheduling/local_resource_manager.h

```diff
@@ -107,6 +108,11 @@ class LocalResourceManager {
 
   void ReleaseWorkerResources(std::shared_ptr<TaskResourceInstances> task_allocation);
 
+  int64_t Version() {
+    UpdateAvailableObjectStoreMemResource();
```

Comment:
It's weird that this is here, is it easy to change it in this PR?

Reply:
I agree that it's wired, but in the current code, we only update the resource when we creat the message
(https://sourcegraph.com/github.com/ray-project/ray/-/blob/src/ray/raylet/scheduling/local_resource_manager.cc?L316:3)


So the previous one is:
- create it (which will update the memory field)

The current one is:
- check whether something changed by checking the versioning (which will update the memory field)
- if something changes, create it.

I don't think it's easy to change. There are several options:
- one thread calling this
- someplace calling this in asio
- plasma store includes the version as well.

The first two are not doing things better than here and I think the last one seems better.

Reply:
Can we move this into the caller (i.e. CreateSyncMessage()), right before calling Version() so Version() can be const?

Reply:
I feel it's even worse. Here the caller is ray syncer.

Reply:
Ok, I think it's doable but I'm not sure whether it's a good idea or not. Think about this Version means you'd like to check the current status and you'll expect the number to be greater if the current status is newer.

This means whenever when the caller wants to check the version, the caller needs to keep in mind to call update memory usage before this.

I feel this makes it worse and easier to make mistakes compared with what we did here. Here version shows that it'll change some internal status, and the caller will notice this by a compiling error. But for the error I mentioned above, the caller won't be able to notice this.

Reply:
I think it should only be one place `std::optional<syncer::RaySyncMessage> LocalResourceManager::CreateSyncMessage(` and I feel it make sense there, when we try to create sync message, we should try to update memory first.

At least, we should try to update the function name to better reflect what it does? e.g., UpdateMemoryAndGetVersion() or something like that.

Reply:
Synced offline and I deleted Version here.